sendgrid/models/email.py

```diff
@@ -26,25 +26,19 @@
 
 class Email(models.Model):
     """ Email message sent through SendGrid """
-    _name = 'sendgrid.email'
-    _order = 'sent_date desc'
+    _inherit = 'mail.mail'
 
     ##########################################################################
     #                                 FIELDS                                 #
     ##########################################################################
-    email_to = fields.Char(required=True)
-    email_from = fields.Char(
-        required=True,
-        default=lambda self: self.get_default_from())
-    cc_address = fields.Char()
-    subject = fields.Char()
-    body_html = fields.Text()
     body_text = fields.Text()
     sent_date = fields.Datetime(copy=False)
     substitution_ids = fields.One2many(
         'sendgrid.substitution', 'email_id', copy=True)
     layout_template_id = fields.Many2one('sendgrid.template')
     text_template_id = fields.Many2one('email.template')
+
+    # @TODO integrate with mail.mail
```

Comment:
I think mail.mail already defined the state field so you can remove it. What you need to ensure is that state goes to `sent` when using your send method.